frontend/src/pages/projects/screens/detail/overview/serverModels/deployedModels/DeployedModelsGallery.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import {
+  Bullseye,
+  Button,
+  Flex,
+  FlexItem,
+  Gallery,
+  Spinner,
+  Text,
+  TextContent,
+} from '@patternfly/react-core';
+import { useNavigate } from 'react-router-dom';
+import { ModelServingContext } from '~/pages/modelServing/ModelServingContext';
+import { InferenceServiceKind } from '~/k8sTypes';
+import { ProjectDetailsContext } from '~/pages/projects/ProjectDetailsContext';
+import DeployedModelCard from './DeployedModelCard';
+
+interface DeployedModelsGalleryProps {
+  showFailed: boolean;
+  showSuccessful: boolean;
+}
+
+const DeployedModelsGallery: React.FC<DeployedModelsGalleryProps> = ({
+  showSuccessful,
+  showFailed,
+}) => {
+  const navigate = useNavigate();
+  const { currentProject } = React.useContext(ProjectDetailsContext);
+  const {
+    servingRuntimes: { data: servingRuntimes, loaded: servingRuntimesLoaded },
+    inferenceServices: { data: inferenceServices, loaded: inferenceServicesLoaded },
+  } = React.useContext(ModelServingContext);
+  const [filteredServices, setFilteredServices] = React.useState<InferenceServiceKind[]>([]);
+  const filteredStates = React.useRef<{ [key: string]: boolean }>({});
```

Comment:
It probably makes no difference here but with a generic key of type `string` the value should be `boolean | undefined`